src/mavoscript.js

```diff
@@ -240,7 +241,18 @@ var _ = Mavo.Script = {
 		},
 		"and": {
 			logical: true,
-			scalar: (a, b) => !!a && !!b,
+			scalar: (a, b) => {
+				if (typeof a !== "boolean" && a !== 0) {
+					return !!b && a;
+				}
+				else if (typeof b !== "boolean" && b !== 0) {
+					return !!a && b;
+				}
+				else {
+					return !!a && !!b;
+				}
+			},
+			unary: () => false,
```

Comment:
Ya, when 1 operand is missing, we decided that it would return false.

I *did* try to set the identity to false. But because of how the `addBinaryOperator` function is implemented, the final result of the operation would never be an array, so that’s why I chose to use unary instead

Reply:
Could you expand on that? Why would it never be an array? Is there a bug?

Reply:
It seems like after an operation is completed, the `addBinaryOperator` function is called and `o.identity` is assigned to `prev` (line 95), which then becomes the operand of `and` on line 111 and makes `result` always false.

<img width="875" alt="screen shot 2018-08-04 at 5 31 50 pm" src="https://user-images.githubusercontent.com/16854657/43681583-62cd79f4-980c-11e8-998f-b5e1927f6ba3.png">

Reply:
That seems rather broken. No idea what I was thinking.

Could you look into fixing the logic there instead of working around it in your code?

It looks like the `logical` flag is not needed for `and` and `or`. It looks like this was mainly created for things like `a < b < c` which you want to expand as `a < b && b < c`, hence the conjunction in line 111. It appears to have already caused trouble with `or`, which is why it has a `reduce` method (it appears to be the ONLY operator with a reduce method, so I suspect the whole point of `o.reduce` was to avoid this issue).

But it's been a while since I touched this code, so do investigate first! And add any tests needed :)

Reply:
I inspected that code block and I believe you are right. I have 2 possible fixes:

1. Remove the logical flag for `and` and `or` since you mentioned they are not needed, and rename `logical` to `comparison` since the flag is only used for comparison operators after removal.
2. Define a `reduce` method for `and` too.

Reply:
The 2 is adding a workaround instead of solving the underlying issue. Avoid doing that unless you're under time pressure. Solving the underlying issue is always preferable and saves time in the long run :)

Can you try 1 and see how it goes?
Also, that should probably be a separate pull request since it can be merged independently.

Reply:
Ya, I don't think 1 broken anything, and we don't need `reduce` anymore. Here's the PR #381